scan.go

```diff
@@ -968,6 +968,11 @@ func (zl *zlexer) Next() (lex, bool) {
 				// was inside braces and we delayed adding it until now.
 				com[comi] = ' ' // convert newline to space
 				comi++
+				if comi >= len(com) {
+					l.token = "comment length insufficient for parsing"
+					l.err = true
+					return *l, true
+				}
```

Comment:
it may be well possible that breaking here would be enough. I have little knowledge of this parsing code to make a good judgement call. Sanity checking here (like is done at the beginning of the for loop) just work (tm)

Reply:
should this be `comi+1` as we increment it again on line 978? think it's fine to error here, instead of breaking and letting the check on line 847 pick it up